extensions/positron-python/src/client/positron/session.ts

```diff
@@ -538,35 +539,45 @@ export class PythonRuntimeSession implements positron.LanguageRuntimeSession, vs
         );
     }
 
-    async activateLsp() {
+    async activateLsp(): Promise<void> {
         // Start LSP for the foreground session only if its been previously stopped
         if (this._lsp?.state === LspState.stopped) {
-            this._queue.add(async () => {
-                const port = await this.adapterApi!.findAvailablePort([], 25);
-                if (this._kernel) {
-                    this._kernel.emitJupyterLog(`Starting Positron LSP server on port ${port}`);
-
-                    // Create the LSP comm before creating the LSP
-                    // client. We keep track of this initialisation in
-                    // case we need to restart, to avoid restarting in
-                    // the middle of init.
-                    this._lspStarting = this._kernel.startPositronLsp(`127.0.0.1:${port}`);
-
-                    await this._lspStarting;
-                    await this._lsp?.activate(port);
-                }
+            return this._queue.add(async () => {
+                await this.startLsp();
```

Comment:
I went to great lengths to reduce duplication here. Now we always call `this.startLsp()` here and when in the `state === positron.RuntimeState.Ready` state, rather than having duplicated code. Similar with `deactivateLsp()`.